### PR TITLE
bulkupload get files and get by id

### PR DESCRIPTION
**Pull request by dbjsdev**
Outstanding issues: integrate in the metadata class once its prior changes are merged.
expiry of signed URL - currently set to same as the upload url, should this be different?

### CODE REVIEW

File: server/routes/establishments/bulkUpload.js [hunk 1]
```diff
@@ -26,7 +26,98 @@ const TrainingEntity = require('../../models/classes/training').Training;
...
+    const params = {
+      Bucket: appConfig.get('bulkuploaduser.bucketname').toString(), 
+      Prefix: `${req.establishmentId}/latest/`
+    };
+
+    const data = await s3.listObjects(params).promise();
+    const returnData = await Promise.all(data.Contents.filter(myFile => !ignoreMetaDataObjects.test(myFile.Key) && !ignoreRoot.test(myFile.Key))
+        .map(async(file) => {
+          const elements = file.Key.split("/"); 
+          const objData = await s3.getObject({ Bucket: params.Bucket, Key: file.Key}).promise();
```
Comment (wozzer72) [inline server/routes/establishments/bulkUpload.js]: Can we do the getObject but just medata? We don't want to be downloading a 3MB file, just to get metadata on the object.

File: server/routes/establishments/bulkUpload.js [hunk 2]
```diff
@@ -26,7 +26,98 @@ const TrainingEntity = require('../../models/classes/training').Training;
...
+            uploaded: file.LastModified,
+            uploadedBy: objData.Metadata.username,
+            records: 0,
+            errors: 0,
+            warnings: 0,
+            type: null,
+            size: file.Size,
+            etag: file.ETag,
+            key: file.Key,
+            signedUrl : s3.getSignedUrl('getObject', {
```
Comment (wozzer72) [inline server/routes/establishments/bulkUpload.js]: So so sorry. I was a tad hasty in drafting the interface for list, but the signedURL is not required for the list response.

File: server/routes/establishments/bulkUpload.js [hunk 3]
```diff
@@ -26,7 +26,98 @@ const TrainingEntity = require('../../models/classes/training').Training;
...
+
+router.route('/uploaded/:etag').get(async (req, res) => {
+  try {
+    const requestedETag = req.params.etag;
+    const params = {
+      Bucket: appConfig.get('bulkuploaduser.bucketname').toString(), 
+      Prefix: `${req.establishmentId}/latest/`
+    };
+    
+    const data = await s3.listObjects(params).promise();
```
Comment (wozzer72) [inline server/routes/establishments/bulkUpload.js]: We shouldn't need to list objects, if we already have the primary key, namely, etag.

File: server/routes/establishments/bulkUpload.js (hunk 1 above)
Comment (dbjsdev) [reply server/routes/establishments/bulkUpload.js]: Yup , change implemented

File: server/routes/establishments/bulkUpload.js (hunk 2 above)
Comment (dbjsdev) [reply server/routes/establishments/bulkUpload.js]: url removed from listing

File: server/routes/establishments/bulkUpload.js [hunk 4]
```diff
@@ -26,7 +26,98 @@ const TrainingEntity = require('../../models/classes/training').Training;
...
+
+    const fileData = data.Contents.filter(myFile => myFile.ETag == '"'+requestedETag+'"' && !ignoreMetaDataObjects.test(myFile.Key) && !ignoreRoot.test(myFile.Key));
+
+    if(fileData.length == 1){
+      const objHeadData = await s3.headObject({ Bucket: params.Bucket, Key: fileData[0].Key}).promise();
```
Comment (wozzer72) [inline server/routes/establishments/bulkUpload.js]: I likely. I likely!!!!!

File: server/routes/establishments/bulkUpload.js [hunk 5]
```diff
@@ -26,7 +26,98 @@ const TrainingEntity = require('../../models/classes/training').Training;
   
 const FileValidationStatusEnum = { "Pending": "pending", "Validating": "validating", "Pass": "pass", "PassWithWarnings": "pass with warnings", "Fail": "fail" };
 
+const ignoreMetaDataObjects = /.*metadata.json$/;
+const ignoreRoot = /.*\/$/;
 
+router.route('/uploaded').get(async (req, res) => {
```
Comment (wozzer72) [inline server/routes/establishments/bulkUpload.js]: We need to refactor this into a it's on router file. We should also take this opportunity to refactor the bulkupload itself. But let's raise this a backlog card.

**Pull request merged**